x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilder.java

```diff
@@ -236,6 +246,11 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) {
                 shouldPruneTokens,
                 tokenPruningConfig
             );
+        } else if (inferenceId == null) {
+            // Edge case, where inference_id was not specified in the request,
+            // but we did not intercept this and rewrite to a query o field with
+            // pre-configured inference. So we trap here and output a nicer error message.
+            throw new IllegalArgumentException("inference_id required to perform vector search on query string");
```

Comment:
Including the field name and the (wrong) field type might help users diagnose the problem

```suggestion
            throw new IllegalArgumentException("inference_id required to perform a sparse_vector query on sparse_vector field [" + fieldName + "]");
```